internal/fsapi/file.go

```diff
@@ -371,12 +366,155 @@ type File interface {
+	//   - syscall.EINVAL: the offset is larger than the current internal cursor.
+	//   - syscall.ENOSYS: the offset is too small for the current window.
+	//   - syscall.EBADF: the file or directory was closed.
+	//
+	// # Notes
+	//
+	//   - This is conceptually similar to invoking Seek(offset, io.SeekStart)
+	//     on io.Seeker and `fseek` in POSIX over an open directory. See
+	//     https://pubs.opengroup.org/onlinepubs/9699919799/functions/fseek.html
+	//   - However implementations will generally keep a buffer of the currently
+	//     valid values and will not actually seek to the given offset the underlying
+	//     open directory.
+	//   - For offset 0, this is conceptually similar to invoking Seek(0, io.SeekStart)
+	//     on io.Seeker and `fseek`, however implementations will generally
+	//     reopen the underlying directory for portability reasons.
+	Rewind(offset uint64) syscall.Errno
+
+	// Peek emits the value currently pointed by the internal cursor.
+	//
+	// # Errors
+	//
+	// A zero syscall.Errno is success. The below are expected otherwise:
+	//   - syscall.ENOENT: there are no more entries to fetch
+	//   - syscall.EBADF: the directory is no longer valid
+	//   - other error values would signal an issue with fetching the next batch of values.
+	//
+	// # Notes
+	//
+	//   - Ultimately, implementations may fetch data using a `readdir`-like API, thus
+	//     errors will reflect file system errors similarly to the POSIX call. See
+	//     https://pubs.opengroup.org/onlinepubs/9699919799/functions/readdir.html
+	Peek() (*Dirent, syscall.Errno)
+
+	// Next advances the internal cursor to the next value.
+	//
+	// This is equivalent to invoking Peek() and then advancing the internal
+	// cursor; thus, the error conditions are the same.
+	//
+	// # Errors
+	//
+	// A zero syscall.Errno is success. The below are expected otherwise:
+	//   - syscall.ENOENT: there are no more entries to fetch
+	//   - syscall.EBADF: the directory is no longer valid
+	//   - other error values would signal an issue with fetching the next batch of values.
+	//
+	// # Notes
+	//
+	//   - Ultimately, implementations may fetch data using a `readdir`-like API, thus
+	//     errors will reflect file system errors similarly to the POSIX call. See
+	//     https://pubs.opengroup.org/onlinepubs/9699919799/functions/readdir.html
+	Next() (*Dirent, syscall.Errno)
+
+	// Close closes the underlying file.
+	//
+	// # Errors
+	//
+	// A zero syscall.Errno is returned if unimplemented or success. The below
+	// are expected otherwise:
 	//   - syscall.ENOSYS: the implementation does not support this function.
+	//   - syscall.EBADF: the file or directory was closed.
 	//
 	// # Notes
 	//
 	//   - This is like syscall.Close and `close` in POSIX. See
 	//     https://pubs.opengroup.org/onlinepubs/9699919799/functions/close.html
 	Close() syscall.Errno
 }
+
+// ReaddirAll reads eagerly all the values returned by the given
```

Comment:
This is only used in tests and gojs which will be removed in less than a half year. Can you either inline it or move it to the sysfs package?